### ISSUE TITLE
Can't reproduce results in Paper for FedU2-BYOL

### ISSUE THREAD

**Issue by sadimanna** [XeniaLLL/FedU2#2]
While the code repo presents a commendable assortment of several Federated learning methods, there remain some questions about the implementation as a beginner in this domain.

First, why the use of `whole_model` parameter in the FedU2 model? Setting it to False throws several errors, and I see that no provision is present in the code for `whole_model=False` to deal with the issues. 

Second, if `whole_model=True` and `mtl_method=EUA`, then each `ByolEvenClient` client model runs for atleast a maximum of 10 epochs as per the code. Why is this different from the details mentioned in the paper? Or is this just for ablation?

Changing `10` to `5` (as recommended in the paper) does not seem to reproduce the results from the paper, or am I doing something wrong or missing something here?

```
if whole_model and mtl_method!='AVG':
    max_sample_num = max([len(m) for m in client_indices])
    max_steps = (max_sample_num // batch_size+1) * local_epochs
else:
    max_steps= -local_epochs
```

In `ByolEvenClient`,
```
if self.local_epochs<0:
    E=-self.local_epochs
else:
    E=min(self.local_epochs // len(self.aug_train_loader), 10)
```

Finally, I observed that the optimizer used in pre-training is set to SGD in the code, whereas in the paper it is written that Adam was used.

Additionally, the gradient being passed to the `mgda` method is `grads/grads.norm(p=2,dim=0)`, where `grads = server_param - client_param`. But as per your mathematical derivations, should the grads be equal to `(server_param - 
 client_param)/(server_param-client-param).square().sum()` because the derivative of $\log(u(w_t)$, i.e. $\nabla \log(u(w_t)) = \frac{\nabla u(w_t)}{u(w_t)}$, where $u(w_t) = \lVert \theta - \theta_t\rVert^2$, right? Also, it is not clear if ADMM has even been used to find the minimum norm element $d^t$.

Any help would be appreciated.

**Comment by XeniaLLL** [comment]
Do you still suffer from the issue mentioned above?
